Sources/Interpreter/Interpreter.swift

```diff
@@ -296,6 +299,54 @@ public struct Interpreter {
     topOfStack.registers[i] as? Address
   }
 
+  /// Returns the value of `x` if it has a address, or `nil` if it does not.
+  func address(_ x: Operand) -> Address? {
+    switch x {
+    case .register(let instruction):
+      return topOfStack.registers[instruction] as? Address
+    case .parameter(_, let i):
+      return topOfStack.parameters[i]
+    case .constant:
+      return nil
+    }
+  }
+
+  /// Returns the address of `subField` in the object  at `origin`.
+  mutating func address(of subField: RecordPath, in origin: Address) -> Address {
+    let l = origin.startLocation;
+    var o = l.offset;
+    var t = origin.type;
+    for i in subField {
+      o += t.parts[i].offset
+      t = typeLayout[t.parts[i].type]
+    }
+    return .init(startLocation: .init(allocation: l.allocation, offset: o), type: t)
```

Comment:
No `;`s. No raw loops.  Reassigning `t` is wasteful.

```suggestion
    let l = origin.startLocation
    let t = origin.type
    let o = subField.reduce(into: l.offset) { p, i in o += t.parts[p].offset })
    return .init(
      startLocation: .init(allocation: l.allocation, offset: o), 
      type: typeLayout[t.parts.last?.type ?? t])
```

If `t.parts` is never empty in this case you can simplify the last line further.  

We should really have a `sum()` algorithm for this.

```suggestion
    let l = origin.startLocation
    let t = origin.type
    let o in subField.lazy.map { p in t.parts[p].offset }.sum(into: origin.startLocation)
    return .init(
      startLocation: .init(allocation: l.allocation, offset: o), 
      type: typeLayout[t.parts.last?.type ?? t])
```